contextIO.go

```diff
@@ -149,5 +171,13 @@ func (c *ContextIO) DoJSON(method, q string, params url.Values, body *string) (j
 	}
 	defer response.Body.Close()
 	json, err = ioutil.ReadAll(response.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	if response.StatusCode != http.StatusOK {
```

Comment:
I am hesitant to put HTTP error codes in as golang errors, response codes could be something other than `StatusOK` but still not be an error in the strict golang sense of the word.  There are cases where a 404 is not a valid response but not an error, and there are 2XX codes that are returned by the API that aren't `StatusOK`.  This should really be the responsibility of the user of the lib to determine.  Golang `errors` should be returned when there is a failure with the request, not an error from the API.